skxray/core.py

```diff
@@ -729,6 +730,118 @@ def wedge_integration(src_data, center, theta_start,
     raise NotImplementedError()
 
 
+def radial_integration(image_array, calibrated_center,
+                       x_axis="r", wavelength=None,
+                       pixel_size=None, dist_sample=None,
+                       num_bins=None):
+    """
+    Radial integration : taking the 2D powder image data
+    and convert to 1D
+
+    Parameters
+    ----------
+    image_array : ndarray
+        input image
+
+    calibrated_center : tuple
+        The center in pixels-units (row, col)
+
+    x_axis : {'r', 'two_theta', 'q'}, optional
+         radius (mm), two_theta (degrees) or
+          q (1/Angstroms)
+
+    wavelength : float, optional
+        wavelength of the incoming x-rays (Angstroms)
+
+    pixel_size : tuple, optional
+        The size of a pixel in real units.
+        (height, width). (mm)
+
+    dist_sample : float, optional
+        distance from the sample to the detector (mm)
+
+    num_bins : int, optional
+        number of bins
+
+    Returns
+    -------
+    bin_centers : ndarray
+       bin centers from bin edges
+
+    ring_average : ndarray
+        radial integration of intensity
+
+    """
+    # convert to pixels to radius
+    pixel_val = pixel_to_radius(np.shape(image_array), calibrated_center)
+
+    if x_axis == 'r':
+        x_val = pixel_val
+    else:
+        x_val = _process_x_val(x_axis, pixel_val, dist_sample, wavelength)
+
+    if num_bins is None:
+        num_bins = np.sum(image_array.shape)
+
+    bin_edges, sums, counts = bin_1D(np.ravel(x_val), np.ravel(image_array),
+                                     nx=num_bins)
+
+    mask = counts > 10
+    # getting bin centers from bin edges
+    bin_centers = bin_edges_to_centers(bin_edges)[mask]
+    # radial integration
+    ring_averages = sums[mask] / counts[mask]
+
+    return bin_centers, ring_averages
+
+
+def _process_x_val(x_axis, val, dist_sample, wavelength):
+    """
+    Parameters
+    ----------
+    x_axis : str
+        {'q', 'two_theta', 'r', 'd'}
+
+    val : ndarray
+        radius values from the calibrated center
+        shape (image array shape)
+
+    dist_sample : float
+        distance from the sample to the detector (mm)
+
+    wavelength : float
+        wavelength of the incoming x-rays (Angstroms)
+
+    Returns
+    -------
+    x_val : ndarray
+        x axis values (two theta or Q space)
+        shape (image array shape)
+
+    """
+    if dist_sample is 'None':
```

Comment:
dist_sample is optional for the radial_integration module. However, to get two theta values or q values you need dist_sample. Therefore, there should be raise valueError here to check it